### PR TITLE
Fix checking for gui updates

### PR DESCRIPTION
**Pull request by mmbyday**
#2089
old

### CODE REVIEW

**Comment by xmrdsc** [comment]
`function onWalletCheckUpdatesComplete(update) {`

update is always empty for me. This is the bug I meant in #2089

**Comment by erciccione** [comment]
I'm not sure the direct link is the best option.Maybe would be better something more mininmalistic? like: "a new version is available, check out getmonero.org".

**Comment by xmrdsc** [comment]
To verify, how were you able to test the update mechanism? Can you confirm the GUI:

1. Makes a DNS request on startup
2. `WalletManager::checkUpdatesAsync` emits the `checkUpdatesComplete` with a non-empty `update` parameter when it is done checking.
3. So that `onWalletCheckUpdatesComplete(update)` in `main.qml` is not empty
4. So that it shows the dialog

^ cannot reproduce

**Comment by mmbyday** [comment]
@xmrdsc Good questions. I tested this PR against the v0.13.0.4 branch, as you may see from clues in the GUI (missing accounts, openalias placeholdertext)

Yes, your understanding sounds almost right. In step 2, `checkUpdatesComplete` can be empty when there is no update.

You can look under the hood of `WalletManager::checkUpdates` and see that the `tuple `of update data comes from `Monero::WalletManager::checkUpdates` here:
https://github.com/monero-project/monero/blob/29073f65e8816d4c32b6ffef514943a5650b8d3b/src/wallet/api/wallet_manager.cpp#L334

When there is no new update, this is why `update` parameter is empty.
```
    if (!tools::check_updates(software, buildtag, version, hash))
      return std::make_tuple(false, "", "", "", "");
```

Hope that helps you.

**Comment by mmbyday** [comment]
@erciccione That's a good option. It's simple and minimalistic as you say. Ultimately however, I think the typical user wants a one-click update though. That's what most user friendly wallets do for updates, like ledger-live.

**Comment by mmbyday** [comment]
Rebased.

File: main.qml [hunk 1]
```diff
@@ -1953,9 +1953,13 @@ ApplicationWindow {
         if (parts.length == 4) {
           var version = parts[0]
           var hash = parts[1]
-          var user_url = parts[2]
-          var auto_url = parts[3]
-          var msg = qsTr("New version of monero-wallet-gui is available: %1<br>%2").arg(version).arg(user_url) + translationManager.emptyString
+          //var user_url = parts[2]
+          //var auto_url = parts[3]
+          var osBuildTag = isMac ? "mac-x64" : isWindows ? "win-x64" : isLinux ? "linux-x64" : "unknownBuildTag"
```
Comment (xiphon) [inline main.qml]: Return or throw a error in the unknown/unrecognized OS case.

Comment (mmbyday) [reply main.qml]: Excellent idea. Thanks.

**Comment by mmbyday** [comment]
Updated to handle unknown/unrecognized OS case with generic message.

**Pull request merged**